psi4/src/psi4/libmints/matrix.h

```diff
@@ -863,16 +863,8 @@ class PSI_API Matrix : public std::enable_shared_from_this<Matrix> {
 
     /// @{
     /// Diagonalizes this, eigvectors and eigvalues must be created by caller.  Only for symmetric matrices.
-    void diagonalize(Matrix* eigvectors, Vector* eigvalues, diagonalize_order nMatz = ascending);
+    void diagonalize(Matrix& eigvectors, Vector& eigvalues, diagonalize_order nMatz = ascending);
     void diagonalize(SharedMatrix& eigvectors, std::shared_ptr<Vector>& eigvalues, diagonalize_order nMatz = ascending);
-    void diagonalize(SharedMatrix& eigvectors, Vector& eigvalues, diagonalize_order nMatz = ascending);
```

Comment:
pausing while we reconsider doing away with the convenience functions

Reply:
I guess having a `(SharedMatrix&, Vector&)` overload would not be a big deal, and it _would_ remove the need for most of the call site modifications in this PR. I am OK with both options.

Reply:
Gentle poke, has a decision been reached? #2686 is mostly done, but stalled until this PR is resolved.

Reply:
I can't recall any discussion on the issue.

Reply:
This PR was part of my motivation for bringing up convenience functions in the conf call. Result is that they're going, so this PR is good as-is.

Reply:
I forgot about that. Glad I said "I can't recall" then!